allegro.h

```diff
@@ -609,11 +616,14 @@ typedef class Serial_write_buffer: public Serial_buffer {
     }
     bool check_buffer(int needed);
     void set_string(const char *s) { 
-        char *fence = buffer + len;
+#ifndef NDEBUG
+        const char *fence = buffer + len;
         assert(ptr < fence);
         // two brackets surpress a g++ warning, because this is an
         // assignment operator inside a test.
-        while ((*ptr++ = *s++)) assert(ptr < fence);
+        while ((*ptr++ = *s++)) {
```

Comment:
Did you do any testing? I just realized that the while is doing the work of this function: *ptr++ = *s++, so this modified version will only work if debugging is on.

Reply:
Not since before the initial review, before adding the `#ifndef NDEBUG`. Thanks for pointing this out. I'll fix that and double check everything.

Reply:
It looks like `Serial_read_buffer.get_string()` would have had the same problem, fixed it there as well. I'll do some testing now